src/app/theme/components/baD3Chart/baD3Chart.service.ts

```diff
@@ -0,0 +1,253 @@
+import { Injectable } from '@angular/core';
+import { d3 } from './baD3Chart.loader';
+
+@Injectable()
+export class BaD3ChartService{
+    public chartSetup:any;
+    private width;
+    private height;
+    private data;
+    private xScale;
+    private x1Scale;
+    private yScale;
+    private xAxis;
+    private yAxis;
```

Comment:
Same for all service variables - do we really need to store them as it adds another complexity level as we need to think of them as in any place in the service we can't be sure what they are now, as they are available in all the service code?